### PR TITLE
jquery.fancytree: fixes for Fancytree, FancytreeNode, FancytreeOptions

### PR DESCRIPTION
**Pull request by evil-shrike**
* Fancytree: added fields widget, $container, focusNode, options; fixed activateKey method signature
* FancytreeNode: added fields span, tr; fixed data field type
* FancytreeOptions: fixed toggleEffect field type

### CODE REVIEW

File: jquery.fancytree/index.d.ts [hunk 1]
```diff
@@ -21,18 +21,21 @@ interface JQuery {
 declare namespace Fancytree {
     interface Fancytree {
         $div: JQuery;
-
+        widget: any; //JQueryUI.Widget;
```
Comment (minestarks) [inline jquery.fancytree/index.d.ts]: Why not add the ```JQueryUI.Widget``` type?

File: jquery.fancytree/index.d.ts [hunk 2]
```diff
@@ -199,7 +202,7 @@ declare namespace Fancytree {
         /** Display name (may contain HTML) */
         title: string;
         /** Contains all extra data that was passed on node creation */
-        data: Object;
+        data: any;
```
Comment (minestarks) [inline jquery.fancytree/index.d.ts]: Why is ```any``` necessary here? Could you add some usage examples in the test file to demonstrate?

File: jquery.fancytree/index.d.ts (hunk 1 above)
Comment (minestarks) [inline jquery.fancytree/index.d.ts]: Could you add the actual ```JQueryUI.Widget``` type?

**Review by minestarks — CHANGES_REQUESTED** [review]

File: jquery.fancytree/index.d.ts (hunk 1 above)
Comment (evil-shrike) [inline jquery.fancytree/index.d.ts]: JQueryUI.Widget is implemented in such a way that some fields like `element` and methods like `destroy` are not accessible.

File: jquery.fancytree/index.d.ts (hunk 2 above)
Comment (evil-shrike) [inline jquery.fancytree/index.d.ts]: It's better to ask "why Object is necessary here" :). Actually `data` is anything associated with a node. 
So if I put some object into it in one place:
```
tree.rootNode.data = { model: someDomainObject }
```
then I'd expect it available in another:
```
let model = tree.rootNode.data.model;
```
but it won't compile:
TS2339: Property 'model' does not exist on type 'Object'.

**Pull request merged**